File: swift/octal.swift
import Foundation

enum Exceptions: Error {
    case IllegalArgumentError(String)
}

func decToOct(dec: Int) throws -> String {
    var dec = dec
    if dec < 0 { throw Exceptions.IllegalArgumentError("The integer dec must be positive!") }
    if dec == 0 { return "0" }
    var res = ""
    while dec > 0 {
        res = "\(dec % 8)\(res)"
        dec /= 8
    }
    return res     
}

func octToDec(oct: String) -> Int {
    var result = 0
    for (index, char) in oct.reversed().enumerated() {
        let part = Int(char.wholeNumberValue ?? 0)
        result += part * Int(pow(8, Double(index)))
    }
    return result
}

do {
    print(try decToOct(dec: 32))
    print(octToDec(oct: "234"))
} catch {
    print(error)
}

File: python/octal.py
"""
This file contains algorithms that do decimal to octal conversions and vice versa.
Important note: The decimal numbers here are positive integers.
"""

def dec_to_oct(dec: int):
    res = ""
    while dec != 0:
        res = str(dec % 8) + res
        dec //= 8
    return res

def oct_to_dec(octal: str):
    result = 0
    for n in range(len(octal) - 1, -1, -1):
        result += int(octal[n]) * 8 ** (len(octal) - 1 - n)
    return result

if __name__ == "__main__":
    import os
    result = dec_to_oct(int(os.getenv("NUMBER")))
    print(result)
    print(oct_to_dec(result))
